packages/medusa/src/loaders/models.ts

```diff
@@ -46,6 +47,27 @@ export default (
     config,
   })
 
+  // Apply the extended models to the core models first to ensure that
+  // when relationships are created, the extended models are used
+  coreModels.forEach((modelPath) => {
+    const loaded = require(modelPath) as
+      | ClassConstructor<unknown>
+      | EntitySchema
+
+    if (loaded) {
+      const name = formatRegistrationName(modelPath)
+      const mappedExtensionModel = modelExtensionsMap.get(name)
+      if (mappedExtensionModel) {
+        const coreModel = require(modelPath)
+        const modelName = upperCaseFirst(
+          formatRegistrationNameWithoutNamespace(modelPath)
+        )
+
+        coreModel[modelName] = mappedExtensionModel
```

Comment:
question: Can we do `loaded[modelName] = mappedExtensionModel` instead and save a file read?

Reply:
done all